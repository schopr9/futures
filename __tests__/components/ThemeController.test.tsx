import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ThemeController from 'app/components/ThemeController';

import * as redux from 'react-redux';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('Theme controller', () => {
  const useSelectorSpy = jest.spyOn(redux, 'useSelector');
  useSelectorSpy.mockReturnValue(false);
  it('initial theme not dark', () => {
    const { toJSON } = render(<ThemeController />);
    expect(toJSON).toMatchSnapshot();
  });

  it.skip('should toggle theme on button click', () => {
    let mockDispatch = jest.fn();
    const useDispatchSpy = jest.spyOn(redux, 'useDispatch');
    useDispatchSpy.mockReturnValue(mockDispatch);
    const { getByTestId } = render(<ThemeController />);
    const input = getByTestId('switch');
    fireEvent.changeText(input);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
