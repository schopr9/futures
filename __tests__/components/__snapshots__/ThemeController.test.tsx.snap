// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme controller initial theme not dark 1`] = `[Function]`;
